pkg/controller/group/rollingupdate.go

```diff
@@ -154,10 +166,54 @@ func (r *rollingupdate) waitUntilQuiesced(pollInterval time.Duration, expectedNe
 	}
 }
 
+// exceedsHealthyThreshold returns true for the Updating threshold is exceeded
+func exceedsHealthyThreshold(updating group_types.Updating, expectedNewInstances int, counts *updatingCount) bool {
```

Comment:
Any reason why `counts` is a pointer?  There are no nil checks in this function.  If it's never nil then maybe we don't need pointer...

Reply:
We want to update the `struct` members, if we don't pass a pointer then a copy is passed and it's not mutated.

I originally had this logic embedded into the `waitUntilQuiesced` function but it got too nested and harder to test.

Reply:
Ah I see this is meant to update some state that is used in the caller...

Another option is to have a new `updatingCount` value as the return value, along with the bool:

```
func exceedsHealthyThreshold(Updating, int, updatingCount) (updatingCount, bool)
```
So the return value is the next state.  
Sorry I am being pedantic -- I generally prefer functions that doesn't have side effects -- taking in input, returning output and not mutating something inside the function (unless the function has a receiver now -- which you can make this a "method" on the `updatingCount` struct....    

Don't know if this makes sense to you.  Your call :)

Reply:
Ahh, that's a great point. I'll change the function to have a receiver:
```
func (counts *updatingCount) exceedsHealthyThreshold(updating group_types.Updating, expectedNewInstances int) bool {
```